R/data-data-utils.R

```diff
@@ -0,0 +1,78 @@
+#' Function runs the `code`, masks the `code` and creates `teal_data` object.
+#' @param input (`list`) containing inputs to be used in the `code`
+#' @inheritParams tm_teal_data
+#'
+#' @return `teal_data` object
+#'
+#' @export
+eval_and_mask <- function(data,
+                          code,
+                          input = list(),
+                          input_mask = list()) {
+  checkmate::assert_list(input)
+  if (inherits(input, "reactivevalues")) {
+    input <- shiny::reactiveValuesToList(input)
+  }
+
+  # evaluate code and substitute input
+  data <- teal.code::eval_code(data, .mask_code(code, args = input))
+  if (inherits(data, "qenv.error")) {
+    return(data)
+  }
+
+  if (identical(ls(data@env), character(0))) {
```

Comment:
In case some objects' names starts with `.`?
```suggestion
  if (identical(ls(data@env, all.names = TRUE), character(0))) {
```